ote/src/cljs/ote/app/controller/place_search.cljs

```diff
@@ -66,6 +67,13 @@
   (process-event [{name :name} app]
     (search (assoc-in app [:place-search :secondary-name] name) false))
 
+  SetDrawControl
+  (process-event [{show? :show? primary? :primary?} app]
+    (if (nil? primary?)
+      (assoc-in (update-in app [:place-search] dissoc :primary?) [:place-search :show?] show?)
```

Comment:
I don't know if this is ok, but I would suggest to change this like this:
(-> app
 ( update-in ...)
(assoc-in ....)
(assoc-in ...)
(assoc-in ...)

Reply:
Fixed.